geoapps/edge_detection/driver.py

```diff
@@ -0,0 +1,355 @@
+        grid_data[np.isnan(grid_data)] = 0
+
+        if np.any(grid_data):
+            # Find edges
+            edges = canny(grid_data, sigma=sigma, use_quantiles=True)
+            shape = edges.shape
+            # Cycle through tiles of square size
+            max_l = np.min([window_size, shape[0], shape[1]])
+            half = np.floor(max_l / 2)
+            overlap = 1.25
+
+            n_cell_y = (shape[0] - 2 * half) * overlap / max_l
+            n_cell_x = (shape[1] - 2 * half) * overlap / max_l
+
+            if n_cell_x > 0:
+                cnt_x = np.linspace(
+                    half, shape[1] - half, 2 + int(np.round(n_cell_x)), dtype=int
+                ).tolist()
+                half_x = half
+            else:
+                cnt_x = [np.ceil(shape[1] / 2)]
+                half_x = np.ceil(shape[1] / 2)
+
+            if n_cell_y > 0:
+                cnt_y = np.linspace(
+                    half, shape[0] - half, 2 + int(np.round(n_cell_y)), dtype=int
+                ).tolist()
+                half_y = half
+            else:
+                cnt_y = [np.ceil(shape[0] / 2)]
+                half_y = np.ceil(shape[0] / 2)
+
+            coords = []
+            for cx in cnt_x:
+                for cy in cnt_y:
+
+                    i_min, i_max = int(cy - half_y), int(cy + half_y)
+                    j_min, j_max = int(cx - half_x), int(cx + half_x)
+                    lines = probabilistic_hough_line(
+                        edges[i_min:i_max, j_min:j_max],
+                        line_length=line_length,
+                        threshold=threshold,
+                        line_gap=line_gap,
+                        seed=0,
+                    )
+
+                    if np.any(lines):
+                        coord = np.vstack(lines)
+                        coords.append(
+                            np.c_[
+                                x[i_min:i_max, j_min:j_max][coord[:, 1], coord[:, 0]],
+                                y[i_min:i_max, j_min:j_max][coord[:, 1], coord[:, 0]],
+                                z[i_min:i_max, j_min:j_max][coord[:, 1], coord[:, 0]],
+                            ]
+                        )
+            if coords:
+                coord = np.vstack(coords)
+                object_lines = coord
+                indices = EdgeDetectionDriver.get_indices(
+                    grid,
+                    window_center_x,
+                    window_center_y,
+                    window_width,
+                    window_height,
+                    window_azimuth,
+                    resolution,
+                    object_lines,
+                )
+                xy = object_lines[indices, :2]
+                if np.any(xy):
+                    vertices = np.vstack(object_lines[indices, :])
+                    cells = (
+                        np.arange(vertices.shape[0]).astype("uint32").reshape((-1, 2))
+                    )
+            return vertices, cells, xy
```

Comment:
Hmm, it looks like we might not need `xy`? Looks like it's just a truncation of `vertices`